configuration/persistence.md

```diff
@@ -7,12 +7,238 @@ title: Persistence
+This makes it very easy to perform actions based upon time.
+Here are some examples:
+
+```java
+Lights.changedSince(now.minusMinutes(2).minusSeconds(30))
+Temperature.maximumSince(now.toDateMidnight)
+Temperature.minimumSince(parse("2012-01-01"))
+PowerMeter.historicState(now.toDateMidnight.withDayOfMonth(1))
+```
+
+The "now" variable can be used for relative time expressions, while "parse()" can define absolute dates and times.
+See the [Jodatime documentation](http://joda-time.sourceforge.net/api-release/org/joda/time/format/ISODateTimeFormat.html#dateTimeParser()) for information on accepted formats for string parsing.
+
+## Startup Behavior
+
+Persistence services and the rule engine are started in parallel.
+Because of this, it is possible that, during an openHAB startup, rules will execute before item states used by those rules have been restored.
+(In this case, those unrestored items have an "undefined" state when the rule is executed.)
+Therefore, rules that rely on persisted item states may not work consistently.
+
+### Workaround 1
+
+A workaround which helps in some cases is to create an item e.g. "delayed_start" that is set to "OFF" at startup and to "ON" some time later (when it can be assumed that persistence has restored all items.
+You then write a rule that restores items from your persistence service after the delay has completed.
+The time of the delay must be determined by experimentation.
+How long you need to wait before changing your "delayed_start" item from "OFF" to "ON" depends upon the size of your home automation project and the performance of your platform.
+
+<!-- TODO: Need a code example for Workaround 1. -->
+
+### Workaround 2
+
+Create `$OPENHAB_CONF/rules/refresh.rules` with the following content (This rule runs only once when openHAB starts):
+
+```sh
+var boolean reloadOnce = Temperature
+rule "Refresh rules after persistence service has started"
+  when System started
+then
+  if(reloadOnce)
+    executeCommandLine("$OPENHAB_CONF/rules/rules_refresh.sh")
+  else
+    println("reloadOnce is false")
+  reloadOnce = false
+end
+```
+
+Create a refresh script `$OPENHAB_CONF/rules_refresh.sh` and make it execuitable (`chmod +x rules_refresh.sh`):
+
+```sh
+#This script is called by openHAB after the persistence service has started
+sleep 5
+cd [full_path_to_openhab_config_directory]/rules
+RULES=`find *.rules |grep -v refresh.rules
+for f in $RULES
+do
+  touch $f
+done
+```
+
+The script waits for five seconds and then touches all `*.rules` files (except `refresh.rules`).
+This causes openHAB to reload all rules (openHAB automatically reloads rules when their creation date/time changes).
+Other rules files may be added on new lines.
+Note - you will have to experiment to find the appropriate sleep value for your specific system.
 
 {% include contribution-wanted.html %}
```

Comment:
Idk if we need this explicit after this huge improvement.

Reply:
Agree - I noticed that it would have been good to delete this once I had posted the PR.  I will delete this call for contributions.